### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by FadieievaA**

### CODE REVIEW

**Comment by FadieievaA** [comment]
sorry, need to work on this one a bit more

File: app/main.py [hunk 1]
```diff
@@ -1,3 +1,63 @@
...
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other: Union[int, float, Distance]) -> Distance:
+        if isinstance(other, Distance):
+            return Distance(self.km + other.km)
+        if type(other) in [float, int]:
```
Comment (Bohdan-Salamakha) [inline app/main.py:19]: remove this redundant check

File: app/main.py [hunk 2]
```diff
@@ -1,3 +1,63 @@
...
+            return Distance(self.km + other)
+
+    def __iadd__(self, other: Union[int, float, Distance]) -> Distance:
+        if isinstance(other, Distance):
+            self.km += other.km
+        if type(other) in [float, int]:
```
Comment (Bohdan-Salamakha) [inline app/main.py:25]: remove this redundant check

File: app/main.py [hunk 3]
```diff
@@ -1,3 +1,63 @@
...
+            self.km += other
+        return self
+
+    def __mul__(self, other: int) -> Distance:
```
Comment (Bohdan-Salamakha) [inline app/main.py:29]: other can be float or Distance too

File: app/main.py [hunk 4]
```diff
@@ -1,3 +1,63 @@
...
+        return Distance(self.km * other)
+
+    def __truediv__(self, other: int) -> Distance:
```
Comment (Bohdan-Salamakha) [inline app/main.py:32]: other can be float or Distance too

File: app/main.py [hunk 5]
```diff
@@ -1,3 +1,63 @@
...
+        return Distance(round(self.km / other, 2))
+
+    def __lt__(self, other: Union[int, float, Distance]) -> bool:
+        if isinstance(other, Distance):
+            return self.km < other.km
+        if type(other) in [float, int]:
```
Comment (Bohdan-Salamakha) [inline app/main.py:38]: remove this redundant check

File: app/main.py [hunk 6]
```diff
@@ -1,3 +1,63 @@
...
+            return self.km < other
+
+    def __gt__(self, other: Union[int, float, Distance]) -> bool:
+        if isinstance(other, Distance):
+            return self.km > other.km
+        if type(other) in [float, int]:
```
Comment (Bohdan-Salamakha) [inline app/main.py:44]: remove this redundant check

File: app/main.py [hunk 7]
```diff
@@ -1,3 +1,63 @@
...
+            return self.km > other
+
+    def __eq__(self, other: Union[int, float, Distance]) -> bool:
+        if isinstance(other, Distance):
+            return self.km == other.km
+        if type(other) in [float, int]:
```
Comment (Bohdan-Salamakha) [inline app/main.py:50]: remove this redundant check

File: app/main.py [hunk 8]
```diff
@@ -1,3 +1,63 @@
...
+            return self.km == other
+
+    def __le__(self, other: Union[int, float, Distance]) -> bool:
+        if isinstance(other, Distance):
+            return self.km <= other.km
+        if type(other) in [float, int]:
```
Comment (Bohdan-Salamakha) [inline app/main.py:56]: remove this redundant check

File: app/main.py [hunk 9]
```diff
@@ -1,3 +1,63 @@
...
+            return self.km <= other
+
+    def __ge__(self, other: Union[int, float, Distance]) -> bool:
+        if isinstance(other, Distance):
+            return self.km >= other.km
+        if type(other) in [float, int]:
```
Comment (Bohdan-Salamakha) [inline app/main.py:62]: remove this redundant check

**Review by Bohdan-Salamakha — CHANGES_REQUESTED** [review]
v1

File: app/main.py (hunk 1 above)
Comment (FadieievaA) [reply app/main.py:19]: could you please explain why it is redundant?

**Pull request closed without merging**